src/identity/caclient.rs

```diff
@@ -54,14 +63,20 @@ impl CaClient {
         let req = IstioCertificateRequest {
             csr,
             validity_duration: 60 * 60 * 24, // 24 hours
-            metadata: Some(Struct {
-                fields: BTreeMap::from([(
-                    "ImpersonatedIdentity".into(),
-                    prost_types::Value {
-                        kind: Some(Kind::StringValue(id.to_string())),
-                    },
-                )]),
-            }),
+            metadata: {
```

Comment:
I think the other issue of doing it at this level is that we end up caching this cert for a bunch of IDs if we try to impersonate

`https://github.com/istio/ztunnel/blob/367188d04ae8d19484cb74fa70067a03c5ffaaaf/src/identity/manager.rs#L341`

Reply:
In serverless environment, node is just a virtual concept, so we don't need to prefech certs for pods on the same "node", It is enough to just fetch cert for the current pod, I will optimize the cert prefetch logic.